test/e2e/atlas/search_nodes_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e || (atlas && search_nodes)
```

Comment:
```suggestion
//go:build e2e || (atlas && search)
```

I would favor a more generic `search` flag. If we then have to split it, we can have atlas && search && nodes